tooling/mcerepkg/internal/customize/customize.go

```diff
@@ -0,0 +1,136 @@
+package customize
+
+import (
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var (
+	operandImageEnvVarPrefix = "OPERAND_IMAGE_"
+	imageRegistryParamName   = "imageRegistry"
+)
+
+type Customizer func(unstructured.Unstructured) (unstructured.Unstructured, map[string]string, error)
```

Comment:
Is this type enforced from outside? Why is it not using pointers? Could save some memory using i.e.:
```
type Customizer func(*unstructured.Unstructured) (map[string]string, error)
```
and update the object in the method